smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelAssembler.java

```diff
@@ -246,15 +247,20 @@ public ModelAssembler addImport(Path importPath) {
 
         if (Files.isDirectory(importPath)) {
             try {
-                Files.walk(importPath)
+                Files.walk(importPath, FileVisitOption.FOLLOW_LINKS)
                         .filter(p -> !p.equals(importPath))
                         .filter(p -> Files.isDirectory(p) || Files.isRegularFile(p))
                         .forEach(this::addImport);
             } catch (IOException e) {
                 throw new ModelImportException("Error loading the contents of " + importPath, e);
             }
         } else if (Files.isRegularFile(importPath)) {
-            stringModels.put(importPath.toString(), () -> IoUtils.readUtf8File(importPath.toString()));
+            try {
+                Path path = importPath.toRealPath();
+                stringModels.put(path.toString(), () -> IoUtils.readUtf8File(path.toString()));
```

Comment:
When storing the models, I thikn this should store the path name as `importPath.toString()` rather than `path.toString()` since we want to refer to the original path as the symlink rather than the target of the symlink.

Minor: I think `IoUtils.readUtf8File(path.toString())` can change to `IoUtils.readUtf8File(path)`.

Reply:
> When storing the models, I thikn this should store the path name as `importPath.toString()` rather than `path.toString()` since we want to refer to the original path as the symlink rather than the target of the symlink.
> 

The issue I see with this is the case when the symlink name is not the same as the target filename (specifically no file extensions exist in the symlink). Using `path.toString()` covers that case. This happens because the all `ModelLoader` validates filename before delegating to next loader. For example in the test `importsSymlinkFileWithAllShapes` I have used a different link name than the actual filename to prove that. If I have the same link name as the filename, there is no issue, otherwise the test fails. I would still prefer to store as `path.toString()` since symlink name could potentially be different than filename. Thoughts?

Reply:
I see your point, but I think of the symlink as an implementation detail that shouldn't carry over into the API or error messages. I would guess that symlinks will usually be directories, and it would be very rare to symlink a single model file. If a single file is symlinked, then I'd be OK with requiring that these files end with ".json" or ".smithy".

Do you have a use case where you need to symlink individual files and can't use a valid extension for Smithy models? Given that you can access the contents of a file by pointing though a symlink, is it necessary to expose the files targeted by the symlink?

Reply:
Our logic for generating symlinks is pretty straight forward and it’s really upto the customer smithy model. Therefore we would not have a use case where we would need to have symlink names that is different to filenames. I just wanted to check this with you with an open source mindset. If there is no such concern then there is no need to expose the files targeted by the symlink, I will make the change.

Reply:
I have updated accordingly